content/common/utils/utils.h

```diff
@@ -0,0 +1,50 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
```

Comment:
Add `C++` guard macro.

Be consistent Unix vs. Windows. Either print the function name for both or none. See a discussion [here](https://stackoverflow.com/questions/15305310/predefined-macros-for-function-name-func) about function name.

The `DIE` macros are very similar for Windows and Linux. I would do some macros such as `print_err()` that are implemented as platform-dependent, and that are called by the macro.

Add `ERR` macros (that may then be integrated into `DIE` macros), as shown [here](https://github.com/razvand/posix-system-programming/blob/master/util/util.h).